src/utils/mapHelper.js

```diff
@@ -19,22 +19,33 @@ export const mapModel = (model, initialModel, mappingRules = []) => {
 	// ]
 	// if you don't need this - just don't pass any value for that parameter
 
-	const result = initialModel != null ? cloneDeep(initialModel) : {};
+	const mapModifiedData = model => {
+		const mapModifiedObj = obj => {
+			if (typeof obj !== "object" || Array.isArray(obj)) return obj;
+			if (obj.wasModified !== undefined) return mapModifiedObj(obj.value);
```

Comment:
better to use != null instaed of !== undefined. In most of contexts null and undefined are pretty the same and you would like to handle both cases.